docs/gallery/howto/dynamic_workflows/autogen/dynamic_graph_builder.py

```diff
@@ -0,0 +1,70 @@
+"""
+===============================================
+Use Graph Builder to create a dynamic workflows
+===============================================
+
+"""
+
+# %%
+# Introduction
+# ============
+# The `Graph Builder` allow user to create also dynamic workflows that can change depending on the input.
+#
+# Load the AiiDA profile.
+
+
+from aiida import load_profile
+
+load_profile()
+
+
+# %%
+# Dynamic workflow dependent on the input
+# =======================================
+# Suppose to run a different task depending on the input. We want to run the add_one task if the number is below 2
+# otherwise we want to run a modulo 2 task. For thet we require to create a context.
+
+from aiida_workgraph import task, WorkGraph
+from aiida.orm import Int
+
+
+@task.calcfunction()
+def add_one(x):
+    return x + 1
+
+
+@task.calcfunction()
+def modulo_two(x):
+    return x % 2
+
+
+@task.graph_builder(outputs=[{"name": "result", "from": "context.out"}])
+def add_modulo(i: Int):
+    wg = WorkGraph()
+    if i.value < 2:
+        task = wg.add_task(add_one, x=i)
+    else:
+        task = wg.add_task(modulo_two, x=i)
+
+    task.set_context({"result": "out"})
+    return wg
+
```

Comment:
This is a duplicate as in the `if.` I would suggest using a for loop inside the graph_builder so that the number of tasks depends on an input value, making the work graph dynamic.

Reply:
I think we have quite a few for loops in the examples too therefore I don't see duplication as a reason to rather use for loops. I feel this example is more educative because you are changing the type of task, you are in different branch of your program depending on your input which might be more intuitive to be dynamic.

But I can also just add both.